pkg/migrator/migrator.go

```diff
@@ -0,0 +1,231 @@
+// Package migrator contains upgrade logic that's needed to upgrade a cluster from
+// openshift 4.1.x to openshift 4.2.0.
+package migrator
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-marketplace/pkg/builders"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	v1 "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v1"
+	v2 "github.com/operator-framework/operator-marketplace/pkg/apis/operators/v2"
+	"github.com/sirupsen/logrus"
+	k8s_errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// NewMigrator returns a migrator that updates/cleans up existing/stale
+// resources, when a Openshift 4.1.x cluster is migrated to version 4.2.x.
+// Specifically, a migrator updates existing subscriptions that had information
+// about catalogSources created during operator installation in a Openshift 4.1
+// cluster, to contain information about the datastore catalogSources instead.
+// The migrator also deletes the stale CSCs that were created during operator
+// installation, as well as the datastore CSCs created by OperatorSources
+// in an Openshift 4.1 cluster.
+func NewMigrator(logger *logrus.Entry, client client.Client) migrator {
+	return migrator{
+		logger: logger,
+		client: client,
+	}
+}
+
+type migrator struct {
+	logger *logrus.Entry
+	client client.Client
+}
+
+// Migrate updates existing subscriptions, deletes the CSCs installed during
+// operator installation as well the stale datastore CSCs when the cluster is
+// migrating from Openshift 4.1.x to Openshift 4.2.x
+func (m *migrator) Migrate() error {
+
+	installedCscs, err := UpdateSubscriptions(m.client, m.logger)
+	if err != nil {
+		return err
+	}
+	deleteInstalledCscs(installedCscs, m.client, m.logger)
+	err = deleteDatastoreCscs(m.client, m.logger)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// UpdateSubscriptions updates the existing Subscriptions'
```

Comment:
@aravindhp I was going for the plural `Subscriptions` as `updateSubscriptions` are updating multiple subscriptions.

Reply:
OK. But please update the doc to say Subscriptions reference installed `CatalogSources`. Also use `Subscriptions` everywhere instead of `subscriptions` in some places.